app/jobs/reinstate_job.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class ReinstateJob < ApplicationJob
+  queue_as :default
+  retry_on Dttp::UpdateTraineeStatus::Error
+
+  def perform(trainee_id)
+    trainee = Trainee.find(trainee_id)
+
+    status = trainee.trn.present? ? DttpStatuses::YET_TO_COMPLETE_COURSE : DttpStatuses::PROSPECTIVE_TRAINEE_TRN_REQUESTED
```

Comment:
Potential race condition here? What if the TRN comes back from DTTP whilst this job is running?